### PR TITLE
Added flag to ignore SSL errors that may be encountered.

### PR DESCRIPTION
**Pull request by metalseargolid**

### CODE REVIEW

**Comment by whiteinge** [comment]
This looks great. Thank you for coding it up! It is an oft-requested addition.

May I trouble you to update this to retain the existing indentation and to just add the new flag?

**Comment by metalseargolid** [comment]
No problem, you can thank PyCharm for that. The formatting should be fixed now :)

**Comment by whiteinge** [comment]
Oh, sorry. The multi-line `ignore_ssl_errors` is great. I meant the indentation changes in `pepper/cli.py`. I'll drop comments on a few of the lines to illustrate.

File: pepper/cli.py [hunk 1]
```diff
@@ -12,7 +11,6 @@
 import ConfigParser
 import getpass
 import time
-
 import pepper
```
Comment (whiteinge) [inline pepper/cli.py]: The line breaks here are intentional.

File: pepper/cli.py [hunk 2]
```diff
@@ -88,8 +93,8 @@ def add_globalopts(self):
         #    help="Redirect the output from a command to a persistent data store")
 
         optgroup.add_option('--fail-if-incomplete', action='store_true',
-            dest='fail_if_minions_dont_respond',
-            help=textwrap.dedent('''\
+                            dest='fail_if_minions_dont_respond',
+                            help=textwrap.dedent('''\
             Return a failure exit code if not all minions respond. This option
```
Comment (whiteinge) [inline pepper/cli.py]: Please retain the same indentation here and the similar changes above and below.

**Comment by whiteinge** [comment]
Also the mode change to setup.py.

**Comment by metalseargolid** [comment]
I went through the diff and restored the original formatting for all whitespace changes in pepper/cli.py, and reverted the mode change.

**Comment by whiteinge** [comment]
Sorry for the slow merge. Thanks a lot for making those changes and for the new addition! :+1:

**Comment by jfroche** [comment]
I know the requests dependency is not an option for you but @metalseargolid commit
https://github.com/saltstack/pepper/commit/bccbcc57ee48a721a274caf042d7526d9cafbdde 
adds a dependency on the new python 2.7.9 ssl interfaces.

```ssl.SSLContext``` is only available in 2.7.9

**Comment by whiteinge** [comment]
Whoops, I completely missed that. Thanks for the catch!

It's been on my radar for a bit to inline this into Pepper for older
Python versions. It applies to 2.6 but I'm not clear if it applies to
older versions as well.

https://pypi.python.org/pypi/backports.ssl/0.0.7

**Pull request merged**